google/cloud/pubsub/internal/tracing_exactly_once_ack_handler_test.cc

```diff
@@ -0,0 +1,219 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+
+#include "google/cloud/pubsub/internal/tracing_exactly_once_ack_handler.h"
+#include "google/cloud/pubsub/subscription.h"
+#include "google/cloud/pubsub/testing/mock_exactly_once_ack_handler_impl.h"
+#include "google/cloud/internal/opentelemetry.h"
+#include "google/cloud/testing_util/is_proto_equal.h"
+#include "google/cloud/testing_util/opentelemetry_matchers.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <gmock/gmock.h>
+#include <opentelemetry/trace/semantic_conventions.h>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::google::cloud::pubsub_testing::MockExactlyOnceAckHandlerImpl;
+using ::google::cloud::testing_util::InstallSpanCatcher;
+using ::google::cloud::testing_util::OTelAttribute;
+using ::google::cloud::testing_util::SpanHasAttributes;
+using ::google::cloud::testing_util::SpanHasInstrumentationScope;
+using ::google::cloud::testing_util::SpanKindIsInternal;
+using ::google::cloud::testing_util::SpanNamed;
+using ::google::cloud::testing_util::SpanWithStatus;
+using ::google::cloud::testing_util::StatusIs;
+using ::testing::AllOf;
+using ::testing::ByMove;
+using ::testing::Contains;
+using ::testing::ElementsAre;
+using ::testing::Return;
+
+pubsub::Subscription const kTestSubscription =
+    pubsub::Subscription("test-project", "test-subscription");
+auto constexpr kTestAckId = "test-ack-id";
+
+std::unique_ptr<pubsub::ExactlyOnceAckHandler::Impl>
+MakeTestExactlyOnceAckHandler(
+    std::unique_ptr<MockExactlyOnceAckHandlerImpl> mock) {
+  EXPECT_CALL(*mock, delivery_attempt()).WillRepeatedly(Return(42));
+  EXPECT_CALL(*mock, ack_id()).WillRepeatedly(Return(kTestAckId));
+  EXPECT_CALL(*mock, subscription()).WillRepeatedly(Return(kTestSubscription));
+  auto span =
+      internal::GetTracer(Options{})->StartSpan("test-subscription subscribe");
```

Comment:
What is wrong with `internal::MakeSpan(...)`? That is how we make spans in most of our code and tests.